### ISSUE TITLE
400 Bad Request (plain HTTP to an SSL-enabled port) via reverse proxy to port 80

### ISSUE THREAD

**Issue by Dennis14e** [ConSol/omd-labs-docker#33]
Hello,

I have a reverse proxy (Traefik) which forwards my requests to port 80 of the container. I access the reverse proxy via HTTPS (port 443). In the container, I have set the Apache mode to "own" via "omd config".
In "own" mode, I assume that the container normally speaks HTTP via port 80, but this does not seem to be the case.

Message in browser:
```
Bad Request
Your browser sent a request that this server could not understand.
Reason: You're speaking plain HTTP to an SSL-enabled server port.
Instead use the HTTPS scheme to access this URL, please.
```

curl from container:
to path / (OK):
```
# curl http://127.0.0.1:80/
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>302 Found</title>
</head><body>
<h1>Found</h1>
<p>The document has moved <a href="http://127.0.0.1/main/">here</a>.</p>
<hr>
<address>Apache/2.4.56 (Debian) Server at 127.0.0.1 Port 80</address>
</body></html>
```

to site /main/ (NOT OK):
```
# curl http://127.0.0.1:80/main/
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>400 Bad Request</title>
</head><body>
<h1>Bad Request</h1>
<p>Your browser sent a request that this server could not understand.<br />
Reason: You're speaking plain HTTP to an SSL-enabled server port.<br />
 Instead use the HTTPS scheme to access this URL, please.<br />
</p>
</body></html>
```

My docker-compose.yml
```yaml
version: '3.7'

services:
  omd:
    build:
      context: .
      pull: true
      dockerfile_inline: |
        FROM consol/omd-labs-debian:latest
      args:
        NEW_SITENAME: main
    restart: unless-stopped
    environment:
      - TZ=Europe/Berlin
    volumes:
      - ./data/local:/opt/omd/sites/main/local.mount
      - ./data/etc:/opt/omd/sites/main/etc.mount
      - ./data/var:/opt/omd/sites/main/var.mount
    networks:
      - traefik-net
      - default
    labels:
      - "traefik.enable=true"
      - "traefik.http.services.omd.loadbalancer.server.port=80"
      - "traefik.http.routers.omd-http.entrypoints=http"
      - "traefik.http.routers.omd-http.rule=Host(`my.host.name`)"
      - "traefik.http.routers.omd-http.middlewares=https-redirect@file"
      - "traefik.http.routers.omd-https.entrypoints=https"
      - "traefik.http.routers.omd-https.rule=Host(`my.host.name`)"
      - "traefik.http.routers.omd-https.middlewares=websocket-header@file"

networks:
  traefik-net:
    external: true
    name: traefik-net
  default:
```

Am I missing something, does something need to be configured differently or is this a bug?

**Comment by sni** [comment]
Apache mode "own" does not change https/ssl configuration. Also the omd apache does not have virtual hosts or anything like that. Either use https or http. And by default this is https.
If you don't want ssl, you need to disable this in your apache configuration.

**Comment by Dennis14e** [comment]
I have checked further.
In the curl example I check port 80, which is the main Apache process and not the omd site. This probably only makes a proxy to the omd Apache process (here port 5000). Apparently only HTTPS is running on port 5000.

Interesting is that no matter how I request via curl to the standard ports 80 and 443, I always get the bad request error (because the main Apache process is probably sending/expecting HTTP?).

If I set the Apache mode back to "ssl", I get the message at port 80 that I should switch to HTTPS, at port 443 it works regularly (because the main Apache process then presumably also sends/expects HTTPS)?

So basically I would have to disable HTTPS in the Apache configuration of the omd site.
What is the best/easiest way to do this?
Or should I accept the situation and simply connect to OMD via HTTPS from the reverse proxy?

**Comment by sni** [comment]
sorry, i was wrong. Setting omd apache to own indeed changes ssl behaviour.
What it does not is restarting the system apache which acts as reverse proxy. So i suggest
directly forwarding to the site apache on port 5000 and set that one to either http or https.
But so far i see no reason to change it and just use ssl.
